### PR TITLE
PR: Add `QtCharts` alias for backward compatibility with 1.x

### LINKED ISSUE #304: Qt6: QtCharts namespace incompatibility
The QtCharts.QtCharts namespace was removed in Qt6, and the contents are now directly in the QtCharts module.
To preserve compatibility with Qt5 code, the below works:
```
qtpy.QtCharts.QtCharts = qtpy.QtCharts
```
Maybe this is worth adding in qtpy itself?

### PR DESCRIPTION
**Pull request by dalthviz**
Fixes #304

Note: Needs a rebase after #313 is merged

### CODE REVIEW

**Review by CAM-Gerlach — CHANGES_REQUESTED** [review]
Because you're importing QtCharts in __init__, this makes the separate QtCharts module a hard dependency of QtPy, instead of just users importing `qtpy.QtCharts`, This also causes our tests to fail on platforms that don't have it by default, on which the test would normally be skipped (aside from the coverage-related failure you fixed in #313 ). Can you do this manipulation inside `qtpy.QtCharts` instead of in `__init__ to resolve this issue? Thanks!

**Comment by CAM-Gerlach** [comment]
@dalthviz Just curious—is there a reason for using the the manual `[WiP]` title "tag" instead of marking as draft?

**Comment by dalthviz** [comment]
> @dalthviz Just curious—is there a reason for using the the manual [WiP] title "tag" instead of marking as draft?

Pure habit basically

File: qtpy/__init__.py [hunk 1]
```diff
@@ -211,6 +211,13 @@ class PythonQtWarning(Warning):
 except (ImportError, PythonQtError):
     pass
 
+try:
+    # QtCharts backward compatibility with QtPy 1.x
+    from . import QtCharts
+    QtCharts.QtCharts = QtCharts
+except (ImportError, PythonQtError):
+    pass
```
Comment (ccordoba12) [inline qtpy/__init__.py:219]: I agree with @CAM-Gerlach: can this manipulation be done in the QtCharts module instead of here?

Comment (CAM-Gerlach) [reply qtpy/__init__.py:219]: :point_up_2: ?

Comment (dalthviz) [reply qtpy/__init__.py:219]: Not sure, I will need to check

**Review by ccordoba12 — APPROVED** [review]
Looks good to me now, thanks @dalthviz!

**Review by CAM-Gerlach — APPROVED** [review]
LGTM, thanks @dalthviz !

File: qtpy/__init__.py (hunk 1 above)
Comment (CAM-Gerlach) [reply qtpy/__init__.py]: Looks good!

**Pull request merged**